core/src/main/scala/com/nvidia/spark/rapids/tool/profiling/ApplicationSummaryInfo.scala

```diff
@@ -136,4 +146,22 @@ class SingleAppSummaryInfoProvider(val app: ApplicationSummaryInfo)
   override def getRapidsJars: Seq[String] = {
     app.rapidsJar.map(_.jar).seq
   }
+
+  override def hasFileCacheSupportedFormat: Boolean = {
+    app.dsInfo.exists { info =>
+      info.format.contains("Parquet")
+    }
+  }
+
+  override def getDistinctLocationPct: Double = {
+    100.0 * app.dsInfo.groupBy(_.location).size / app.dsInfo.size
+  }
+
+  override def getRedundantReadSize: Long = {
+    app.dsInfo.groupBy(_.location)
```

Comment:
Nit: `getRedundantReadSize` and  `getDistinctLocationPct` both use `app.dsInfo.groupBy(_.location)`. It is possible to calculate it once and share between the two methods.

Reply:
How can I do that?  I don't want to call the groupBy(_.location) until I need it in AutoTuner.

Reply:
This can be done by defining a field as `lazy`, but you can ignore that for this PR if it is too much.

> A lazy value definition evaluates its right hand side 𝑒 the first time the value is accessed.
scala2.12 documentation https://www.scala-lang.org/files/archive/spec/2.12/04-basic-declarations-and-definitions.html
Scala 3 promises that it is going to be even improved 

```
class SingleAppSummaryInfoProvider(val app: ApplicationSummaryInfo)
  extends AppSummaryInfoBaseProvider {
  private lazy val distinctLocations = app.dsInfo.groupBy(_.location)
  // rest of code here
  override def getDistinctLocationPct: Double = {
    100.0 * distinctLocations.size / app.dsInfo.size
  }

  override def getRedundantReadSize: Long = {
    distinctLocations
      .filter { case(_, objects) => objects.size > 1 }
      .mapValues(_.map(_.data_size).sum)
      .values
      .sum
  }
}
```

Reply:
Added lazy val in latest commit.  Thanks for the tip!